PipeRobot[] robots
PipeRobot simulated_robot
PipeRobot predicted_robot
bool simulated_robot_set


